### ISSUE TITLE
Zero config for sign up attributes breaks with aws-amplify v6 & @aws-amplify/ui-react-native v2

### ISSUE THREAD

**Issue by hanna-becker** [aws-amplify/amplify-ui#4863]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React Native

### Which UI component?

Authenticator

### How is your app built?

expo

### Which region are you seeing the problem in?

eu-central-1

### Please describe your bug.

After upgrading @aws-amplify/ui-react-native to v2.0.6 and aws-amplify to v6.0.8 and applying all the steps from the[ authenticator setup instructions](https://ui.docs.amplify.aws/react-native/connected-components/authenticator), the authenticator cannot automatically infer given_name and family_name as required sign up attributes any longer. This worked in my previous setup with aws-amplify v5 and  @aws-amplify/ui-react-native v1. The fields are now missing in the signup form, unless specified explicitly via the Authenticator component prop: `signUpAttributes={['family_name', 'given_name']}`. 

The [docs](https://ui.docs.amplify.aws/react-native/connected-components/authenticator/configuration#sign-up-attributes) claim that these two attributes are among the ones that are automatically inferred, so specifying them as signUpAttributes shouldn't be necessary.

### What's the expected behaviour?

given_name and family_name are inferred automatically as required sign up attributes from the UserPool and the sign up form has fields for them.

### Help us reproduce the bug!

1. Create an amplify react native app with given_name and family_name as required sign up attributes.
2. Set up the authenticator connected component as specified in docs.
3. Notice that given_name and family_name fields do not appear automatically in the sign up form, unless explicitly passed via `signUpAttributes`.

### Code Snippet

```typescript
import React from 'react';
import { Button } from 'react-native';

import { Amplify } from 'aws-amplify';
import { Authenticator, useAuthenticator } from '@aws-amplify/ui-react-native';

import awsExports from './aws-exports';
Amplify.configure(awsExports);

function SignOutButton() {
  const { signOut } = useAuthenticator();
  return <Button title="Sign Out" onPress={signOut} />;
}

function App() {
  return (
    <Authenticator.Provider>
      <Authenticator               
        // We shouldn't need to pass these
        signUpAttributes={['family_name', 'given_name']}
      >
        <SignOutButton />
      </Authenticator>
    </Authenticator.Provider>
  );
}

export default App;

```

**Comment by calebpollman** [comment]
@hanna-becker Thanks for raising this. We have a fix for the issue [here](https://github.com/aws-amplify/amplify-ui/pull/4861) which will go out in our next release. 

In the meantime you can either pin your `aws-amplify` version to `6.0.5`, or point your `@aws-amplify/ui-react` version to `next`.

**Comment by calebpollman** [comment]
This issue has been resolved in version `^2.0.7` of `@aws-amplify/ui-react-native`

**Comment by hanna-becker** [comment]
Great! It's working as expected now.
